phalerts.py

```diff
@@ -142,32 +142,30 @@ def find_task(title, projects):
         projects: (list of strings) names of projects that should be assigned to
             a task to be returned. Can be empty.
     """
-    project_phids = {find_project_phid(n) for n in projects}
+    project_phids = list({find_project_phid(n) for n in projects})
 
     result = phab_request(
         phab.maniphest.search,
         constraints=dict(
             query='title:"%s"' % title,
             statuses=["open"],
-            projects=projects,
+            projects=project_phids,
         ),
         attachments=dict(projects=True),
         # this expands to "title, id", so we'll get the newest open task if
         # multiple open tasks exist.
         order="title",
     )
     if result["cursor"]["after"]:
-        raise Error("Unexpected 'after' cursor while searching for task %s" %
-                    title)
+        raise Error(
+            "Unexpected 'after' cursor while searching for task %s" % title
+        )
     for task in result["data"]:
         # 'query' constraint actually does a full text search, so we iterate
         # over returned tasks looking for a one with the title we need.
         if task["fields"]["name"] != title:
             continue
-        task_projects = set(task["attachments"]["projects"]["projectPHIDs"])
-        # Check that the task has all required projects assigned to it.
-        if not project_phids.issubset(task_projects):
```

Comment:
I believe you will need to keep this check, since the `projects` constraint will match not only phids, but project names as well. This means that the search might return tasks from other projects (for example, if one of the project phids passed is also part of a different, unrelated project's name).